### PR TITLE
Magic Bounce and Rebound should not use old protocol messages

### PR DESCRIPTION
**Pull request by urkerab**
Reported [here](https://www.smogon.com/forums/posts/7995630). This is going to need client-side changes too (both for the [Pokémon's Magic Bounce] text and the ability activation animation) but this will simplify the necessary changes (I think the old code had to special-case Magic Bounce and Rebound to achieve this).

### CODE REVIEW

**Comment by Zarel** [comment]
I would do that first.

**Comment by urkerab** [comment]
Currently `kwArgs.from` only contains the name of the effect, but also sending the type of the effect means that the standard `activateAbility` and `maybeAbility` could be used to do the heavy lifting.

The other options are b) don't change the server and special-case those two names c) change the server but support both approaches in the client.

**Comment by urkerab** [comment]
I think Protean and Colo(u)r Change have a similar issue, in that their `kwArgs.from` again only contains the name of the effect, and this used to be special-cased so that it would activate as an ability.

**Comment by Zarel** [comment]
You can hardcode it. That's what I've been doing.

**Comment by Zarel** [comment]
I've been meaning to replace the current hardcode system with a new one in `lineParse`, though.

File: data/scripts.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ let BattleScripts = {
 
 		let movename = move.name;
 		if (move.id === 'hiddenpower') movename = 'Hidden Power';
-		if (sourceEffect) attrs += '|[from]' + this.getEffect(sourceEffect);
+		if (sourceEffect) attrs += '|[from] ' + sourceEffect.fullname;
```
Comment (Zarel) [inline data/scripts.js]: I've been trying to remove all the spaces from the protocol, actually.

Comment (Zarel) [reply data/scripts.js]:
```suggestion
		if (sourceEffect) attrs += `|[from]${sourceEffect.fullname}`;
```

**Comment by urkerab** [comment]
Code was added to the client to special-case Magic Bounce but surely the server shouldn't send old protocol messages if it doesn't have to.

File: data/scripts.js (hunk 1 above)
Comment (urkerab) [reply data/scripts.js]: I merged this in as I was rebasing anyway.

**Comment by Zarel** [comment]
Yep, thanks!

**Pull request merged**